salt/utils/yamlloader.py

```diff
@@ -24,27 +26,46 @@ class SaltYamlSafeLoader(BaseLoader):
     to make things like sls file more intuitive.
     """
 
+    @classmethod
+    def remove_implicit_resolver(cls, tag):
+        """Remove a previously registered implicit resolver for a tag."""
+        cls.yaml_implicit_resolvers = {
+            first_char: [r for r in resolver_list if r[0] != tag]
+            for first_char, resolver_list in cls.yaml_implicit_resolvers.items()
+        }
+
     def __init__(self, stream, dictclass=dict):
         super().__init__(stream)
-        if dictclass is not dict:
-            # then assume ordered dict and use it for both !map and !omap
-            self.add_constructor("tag:yaml.org,2002:map", type(self).construct_yaml_map)
-            self.add_constructor(
-                "tag:yaml.org,2002:omap", type(self).construct_yaml_map
-            )
-        self.add_constructor("tag:yaml.org,2002:str", type(self).construct_yaml_str)
-        self.add_constructor(
-            "tag:yaml.org,2002:python/unicode", type(self).construct_unicode
-        )
-        self.add_constructor("tag:yaml.org,2002:timestamp", type(self).construct_scalar)
         self.dictclass = dictclass
 
     def construct_yaml_map(self, node):
+        if self.dictclass is dict:
+            return (yield from super().construct_yaml_map(node))
         data = self.dictclass()
         yield data
         value = self.construct_mapping(node)
         data.update(value)
 
+    def construct_yaml_omap(self, node):
+        # BaseLoader.construct_yaml_omap() returns a list of (key, value)
+        # tuples, which doesn't match the semantics of the `!!omap` YAML type.
+        # Convert the list of tuples to an OrderedDict.
+        d = collections.OrderedDict()
+        yield d
+        (entries,) = super().construct_yaml_omap(node)
+        if hasattr(entries, "keys"):
+            entries = ((k, entries[k]) for k in entries.keys())
+        for k, v in entries:
+            if k in d:
+                raise ConstructorError(
+                    f"while constructing an ordered map",
+                    node.start_mark,
+                    f"duplicate key encountered: {k!r}",
+                    # TODO: Can we get the location of the duplicate key?
+                    node.start_mark,
+                )
+            d[k] = v
```

Comment:
```suggestion
        inter_d_entries = set(entries).intersection(d)
        if len(inter_d_entries) > 0:
            raise ConstructorError(
                f"while constructing an ordered map",
                node.start_mark,
                f"duplicate keys encountered: {inter_d_entries}",
                node.start_mark,
            )
        d.update(entries)
```

don't need the loop and can report all of the duplicates in a single exception.

Reply:
The goal is to identify duplicates in `entries`, not to identify overlap between `entries` and `d`.  Note that `d` is empty at the time `set(entries).intersection(d)` would be called, so the condition would always be false.